File: 백준/Silver/14405. 피카츄/피카츄.cc
#include <bits/stdc++.h>

using namespace std;

string s;
bool flag;
int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL); cout.tie(NULL);
    cin >> s;
    for(int i = 0; i < s.size(); i++) {
        if(i < s.size() - 1 && (s.substr(i, 2) == "pi" || s.substr(i, 2) == "ka")) i++;
        else if(i < s.size() - 2 && s.substr(i, 3) == "chu") i += 2;
        else {
            flag = 1;
            break;
        }
    }
    cout << ((flag == 1) ? "NO" : "YES") << '\n';
    return 0;
}

File: 백준/Silver/14405. 피카츄/피카츄.swift

import Foundation

let s = Array(readLine()!)
var flag = false

var i = 0
while i < s.count {
    if i <= s.count - 2 && (String(s[i..<i+2]) == "pi" || String(s[i..<i+2]) == "ka") {
        i += 2
    } else if i <= s.count - 3 && String(s[i..<i+3]) == "chu" {
        i += 3
    } else {
        flag = true
        break
    }
}

print(flag ? "NO" : "YES")